roles/kubernetes/client/tasks/main.yml

```diff
@@ -61,11 +61,30 @@
     rm -rf {{ kube_config_dir }}/external_kubeconfig
   environment: "{{ proxy_env }}"
   run_once: yes
-  register: admin_kubeconfig
+  register: raw_admin_kubeconfig
+
+- name: Convert kubeconfig to YAML
+  set_fact:
+    admin_kubeconfig: "{{ raw_admin_kubeconfig.stdout | from_yaml }}"
+
+- name: Extract current settings from kubeconfig
+  set_fact:
+    cluster_infos: "{{ admin_kubeconfig['clusters'][0]['cluster'] }}"
+    user_certs: "{{ admin_kubeconfig['users'][0]['user'] }}"
+    username: "kubernetes-admin-{{ cluster_name }}"
+    context: "kubernetes-admin-{{ cluster_name }}@{{ cluster_name }}"
+
+- name: Override username in kubeconfig
+  set_fact:
```

Comment:
`when: kubeconfig_localhost|default(false)` might be suitable for this task.

Reply:
default(false) is a bit overkill:
```
roles/kubernetes/client/defaults/main.yml:2:kubeconfig_localhost: false
```

Reply:
btw, @mattymo's workaround `Generate admin kubeconfig with external api endpoint` doesn't have the `when` clause, should I add it? It's not use anywhere else

Reply:
probably yes